sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py

```diff
@@ -239,17 +256,36 @@ async def send_batch(
                 :caption: Asynchronously sends event data
 
         """
+        partition_id = kwargs.get("partition_id")
+        partition_key = kwargs.get("partition_key")
+        if isinstance(event_data_batch, EventDataBatch):
+            if partition_id or partition_key:
+                raise TypeError("partition_id and partition_key should be None when sending an EventDataBatch "
+                                "because type EventDataBatch itself may have partition_id or partition_key")
+            to_send_batch = event_data_batch
+        elif isinstance(event_data_batch, List):
+            to_send_batch = await self.create_batch(partition_id=partition_id, partition_key=partition_key)
+            for event_data in event_data_batch:
+                try:
+                    to_send_batch.add(event_data)
+                except ValueError:
+                    raise ValueError("The list of EventData exceeds the Event Hub frame size limit. "
+                                     "Please send a smaller list of EventData, or use EventDataBatch, "
+                                     "which is guaranteed to be under the frame size limit")
```

Comment:
same to the sync suggestion, could moving this part to EventDataBatch object.

Reply:
same as async

Reply:
did the same as in sync code